dynamo-future-file-extension/extension/Extension.cs

```diff
@@ -0,0 +1,128 @@
+
+            var originalOpenFileFromPath = AccessTools.Method(typeof(DynamoModel), nameof(DynamoModel.OpenFileFromPath));
+            var prefix = typeof(FutureFileExtension).GetMethod(nameof(Prefix), System.Reflection.BindingFlags.Static | System.Reflection.BindingFlags.Public);
+            openxmlmethod = typeof(DynamoModel).GetMethod("OpenXmlFileFromPath", System.Reflection.BindingFlags.NonPublic | BindingFlags.Instance);
+            openJsonFileMethod = typeof(DynamoModel).GetMethod("OpenJsonFileFromPath", System.Reflection.BindingFlags.NonPublic | BindingFlags.Instance);
+            if(originalOpenFileFromPath != null && prefix != null && openxmlmethod != null && openJsonFileMethod != null)
+            {
+                harmony.Patch(originalOpenFileFromPath, new HarmonyMethod(prefix));
+            }
+        }
+        public static bool Prefix(DynamoModel __instance, string filePath, bool forceManualExecutionMode = false)
+        {
+
+            __instance?.Logger?.Log($"openfilepath has been patched from{Assembly.GetExecutingAssembly().FullName }");
+            var knownInputTypeNames = Enum.GetValues(typeof(NodeInputTypes)).Cast<NodeInputTypes>().Select(x => ToEnumString(x));
+            XmlDocument xmlDoc;
+            Exception ex;
+            if (DynamoUtilities.PathHelper.isValidXML(filePath, out xmlDoc, out ex))
+            {
+                openxmlmethod.Invoke(__instance, new object[] { xmlDoc, filePath, forceManualExecutionMode });
+                return false;
+            }
+            else
+            {
+                // These kind of exceptions indicate that file is not accessible 
+                if (ex is IOException || ex is UnauthorizedAccessException)
+                {
+                    throw ex;
+                }
+                if (ex is System.Xml.XmlException)
+                {
+                    // XML opening failure can indicate that this file is corrupted XML or Json
+                    string fileContents;
+                    if (DynamoUtilities.PathHelper.isValidJson(filePath, out fileContents, out ex))
+                    {
+                        //at this point filecontents contains a json string - 
+                        //lets parse it, do our transformation and pass it to open method.
+
+                        // TODO we could do more complex replacements, for example, we could replace the WorkspaceConverter.ReadJson method 
+                        // and completely alter how nodeInputs are deserialized. This is just a POC to fix the issue with minimum effort.
+
+                        var root = JObject.Parse(fileContents);
+                        //find all inputs that don't exist in the known types - these are the ones we can't deserialize correctly.
+                        //TODO reason about case sensitivity.
+                        var matchingTokens = root.SelectTokens("Inputs[*].Type").Where(x => !knownInputTypeNames.Contains(x.Value<string>()));
```

Comment:
Shouldn't this be unmatched tokens - tokens that are not found in the current Dynamo version?